scriptworker/worker.py

```diff
@@ -12,23 +12,25 @@
 import os
 import sys
 import signal
+import types
 
 from scriptworker.artifacts import upload_artifacts
 from scriptworker.config import get_context_from_cmdln
 from scriptworker.constants import STATUSES
 from scriptworker.cot.generate import generate_cot
 from scriptworker.cot.verify import ChainOfTrust, verify_chain_of_trust
 from scriptworker.gpg import get_tmp_base_gpg_home_dir, is_lockfile_present, rm_lockfile
-from scriptworker.exceptions import ScriptWorkerException
+from scriptworker.exceptions import ScriptWorkerException, WorkerShutdownDuringTask
 from scriptworker.task import claim_work, complete_task, prepare_to_run_task, \
     reclaim_task, run_task, worst_level
-from scriptworker.utils import cleanup, rm
+from scriptworker.task_process import TaskProcess
+from scriptworker.utils import cleanup, rm, filepaths_in_dir
 
 log = logging.getLogger(__name__)
 
 
 # do_run_task {{{1
-async def do_run_task(context):
+async def do_run_task(context, run_cancellable, to_cancellable_process):
```

Comment:
We'll need to update the docstring here.

Reply:
Surprised `flake8` hasn't caught that - maybe it will, once tests pass